File: Poseidon_SpringAPI/src/main/java/fr/orionbs/models/CurvePoint.java
package fr.orionbs.models;

import lombok.*;

import javax.persistence.*;
import java.sql.Timestamp;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
@Entity(name = "curve_point")
public class CurvePoint {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    private Integer curveId;
    private Timestamp asOfDate;
    private Double term;
    private Double value;
    private Timestamp creationDate;

    @Override
    public String toString() {
        return "CurvePoint{" +
                "id=" + id +
                ", curveId=" + curveId +
                ", asOfDate=" + asOfDate +
                ", term=" + term +
                ", value=" + value +
                ", creationDate=" + creationDate +
                '}';
    }
}

File: Poseidon-AngularFront/src/app/curvePoint/curve-point.ts
export class CurvePoint {
	id?: number;
    curveId?: number;
	term?: number;
	value?: number;
	
	constructor(curveId?: number, term?: number, value?: number) {
		this.curveId = curveId;
		this.term = term;
		this.value = value;
	}

}
